fix(pins): fix max 3 pins and unpining pins the new message

jrainville opened the pull request:
Fixes #4576

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 03b7189f | [#1](https://ci.status.im/job/status-desktop/job/branches/job/macos/job/PR-4651/1/) | 2022-02-01 16:28:11 | ~8 min | `macos` | [:package:`dmg`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-220201-161933-03b718-pr4651.dmg) |

anastasiyaig commented:
@jrainville please apply this pin limit to 1x1 chats and groups as well, since it should be applicable to all the chats where pinning is supported

jrainville commented:
@anastasiyaig Alright, it is fixed. The problem was that the new pinned/unpinned message signal only came for the community. 
I fixed it by re-adding the code I removed, but only modifying the number of pins when the signal is from someone else, because we already modify the number of pins when we do it ourselves

anastasiyaig commented:
@jrainville 

can we handle the response for unpining from mobile as well please please please? So when i do unpin from mobile, desktop wont crash

```
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4651/src/nim_status_client.nim(145) nim_status_client
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4651/src/nim_status_client.nim(142) mainProc
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4651/vendor/nimqml/src/nimqml/private/qguiapplication.nim(29) exec
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4651/vendor/nimqml/src/nimqml/private/qobject.nim(49) qobjectCallback
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4651/vendor/nimqml/src/nimqml/private/qobject.nim(34) onSlotCalled
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4651/vendor/nimbus-build-system/vendor/Nim/lib/core/macros.nim(566) onSlotCalled
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4651/src/app/core/signals/signals_manager.nim(56) receiveSignal
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4651/src/app/core/signals/signals_manager.nim(53) processSignal
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4651/src/app/core/eventemitter.nim(46) emit
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4651/src/app_service/service/message/service.nim(196) :anonymous
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4651/src/app_service/service/message/service.nim(173) handlePinnedMessagesUpdate
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4651/vendor/nimbus-build-system/vendor/Nim/lib/pure/collections/tables.nim(262) []
Error: unhandled exception: key not found: 0x04415706862805e4dad4a85d86519d2759c42375b2ef3738583575a3b67051047459b243614897a046fccd6138cd10c01d9341299e9247090d286ed2c72058a592 [KeyError]
```

jrainville commented:
@anastasiyaig nice find. There was a bug with the chatId vs localChatId in 1-1 chats.

Hopefully, everything should be fixed with pins now 😅

saledjenic commented on src/app_service/service/message/service.nim:
```diff
@@ -167,12 +168,21 @@ QtObject:
       if(pm.pinned):
-        self.numOfPinnedMessagesPerChat[pm.chatId] = self.getNumOfPinnedMessages(pm.chatId) + 1
+        if (chatId != "" and pm.pinnedBy != singletonInstance.userProfile.getPubKey()):
+          self.numOfPinnedMessagesPerChat[chatId] = self.numOfPinnedMessagesPerChat[chatId] + 1
```
@jrainville I deliberately added `getNumOfPinnedMessages`, cause using it, we are safe to get the number of pinned messages for a chat regardless `chatId` is added as a key to the table or not (only because of this check there `if(self.numOfPinnedMessagesPerChat.hasKey(chatId)):`). So would good (safer) to call `getNumOfPinnedMessages` instead of direct access. The same for the line 185.

jrainville replied:
Agreed. You added it after I started my own PR, and yours got merged first. That's why I have a conflict. I'll fix the conflict and use your function

saledjenic reviewed (approved):
@jrainville looks good.
